### ISSUE TITLE
Rule: Warn when initializing to undefined

### ISSUE THREAD

**Issue by nzakas** [eslint/eslint#31]
This rule should warn when a variable is initialized to undefined:

``` js
var foo = undefined;

// same as
var foo;
```

The rule ID should be "no-undef-init" and should be a warning by default.
